Sources/_StringProcessing/Utility/RegexFactory.swift

```diff
@@ -20,6 +20,16 @@ public struct _RegexFactory {
   // Hide is behind an SPI that only RegexBuilder can use.
   @_spi(RegexBuilder)
   public init() {}
+
+  @available(SwiftStdlib 5.8, *)
+  public func ignoreCapturesInTypedOutput(
+    _ child: some RegexComponent
+  ) -> Regex<Substring> {
+    // Don't wrap `child` again if it's a leaf node.
+    child.regex.root.hasChildNodes
+      ? .init(node: .ignoreCapturesInTypedOutput(child.regex.root))
+      : .init(node: child.regex.root)
+  }
```

Comment:
Why is this has-child instead of has-capture? Do we need to wrap this around everything, or is this just part of how capture concatenation works?

Reply:
This branch isn't technically necessary, but it reduces the number of times we apply this wrapper node. I didn't want to do a `hasCapture` check, since that would have to search the whole tree. (But maybe it would be worth it to produce a smaller tree.)

Reply:
Hmm... one of the downsides of using enums for trees is that it's really hard to just add a bit to track `hasCapture` information. 

This looks fine to me for now, the extra nodes in the tree should be transient, though we might want to also track the time spent allocating each node.